docs/appkit/shared/smart-sessions.mdx

```diff
@@ -0,0 +1,269 @@
+
+## Implementations
+
+The `grantPermissions` method provides an easy way to interact with the smart wallet to request permissions.
+
+### Step 1 | Import the method
+First, import the grantPermissions method from the `@reown/appkit-experimental/smart-session` package.
+
+```javascript
+import { grantPermissions, type SmartSessionGrantPermissionsRequest } from '@reown/appkit-experimental/smart-session'
+```
+
+### Step 2 | Define the Permission Request
+
+Create an object adhering to the `SmartSessionGrantPermissionsRequest` type. This object specifies details like the `address`, `chainID`, `signer`, `policies`, `permissions`, and `expiry` time.
+    
+Example request object:
+
+```tsx
+const request: SmartSessionGrantPermissionsRequest = {
+      expiry: Math.floor(Date.now() / 1000) + 24 * 60 * 60, // 24 hours
+      chainId: toHex(baseSepolia.id),
+      address: address,
+      signer: {
+        type: 'keys',
+        data: {
+          keys :[{
+          type: 'secp256k1',
+          publicKey: '0x...' //public key of dapp signer
+        }]
+        }
+      },
+      permissions: [ {
+        type: 'contract-call',
+        data: {
+          address: '0x2E65BAfA07238666c3b239E94F32DaD3cDD6498D', // sample donut contract address
+          abi: [
+            {
+              inputs: [{ internalType: 'uint256', name: 'amount', type: 'uint256' }],
+              name: 'purchase',
+              outputs: [],
+              stateMutability: 'payable',
+              type: 'function'
+            }
+          ],
+          functions: [ {
+            functionName: 'purchase'
+          } ]
+        }
+      }],
+      policies: []
+    }
+```
+
+### Step 3 | Invoke the Method
+
+Call the `grantPermissions` function, passing the request object. This will trigger the permission request via the connected wallet.
+
+```tsx
+ const response = await grantPermissions(request)
+```
+
+### Step 4 | Handle the Response
+
+Upon successful execution, the response will include the granted permissions and the session context. So the response can be handled as needed.
+
+#### Response Format
+
+```tsx
+{
+	chainId: `0x14a34`
+  address: `0x...`
+  expiry: 1727824386
+  permissions: [
+    {
+      type: 'contract-call',
+      data: {
+        address: '0x2E65BAfA07238666c3b239E94F32DaD3cDD6498D', // sample donut contract address
+        abi: [
+          {
+            inputs: [{ internalType: 'uint256', name: 'amount', type: 'uint256' }],
+            name: 'purchase',
+            outputs: [],
+            stateMutability: 'payable',
+            type: 'function'
+          }
+        ],
+        functions: [ {
+          functionName: 'purchase'
+        } ]
+      }
+    }
+  ],
+  context: '...'  // Context identifier for the session
+}
+```
+
+## How to use the permissions
+
+The dApp must call the following two API endpoints from the wallet services to use these permissions.
```

Comment:
Is the `wallet services` API described somewhere else? Else we'd need to specify what this means.